File: 9. JS Front-End/JS Syntax Fundamentals/Vacation.js
function solve(numberOfPeople, typeOfGruop, dayOfWeek) {
    let totaPrice = 0;
    let discount = 0;
    let price = 0;

    switch (typeOfGruop) {
        case 'Students':
            if (dayOfWeek === 'Friday') {
                if (numberOfPeople >= 30) {
                    totaPrice = numberOfPeople * 8.45;
                    discount = totaPrice * (15 / 100);
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 8.45;
                }
            } else if (dayOfWeek === 'Saturday') {
                if (numberOfPeople >= 30) {
                    totaPrice = numberOfPeople * 9.80;
                    discount = totaPrice * (15 / 100);
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 9.80;
                }
            } else if (dayOfWeek === 'Sunday') {
                if (numberOfPeople >= 30) {
                    totaPrice = numberOfPeople * 10.46;
                    discount = totaPrice * (15 / 100);
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 10.46;
                }
            }
            break;
        case 'Business':
            if (dayOfWeek === 'Friday') {
                if (numberOfPeople >= 100) {
                    totaPrice = numberOfPeople * 10.90;
                    discount = (numberOfPeople * 10.90) / 10;
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 10.90;
                }
            } else if (dayOfWeek === 'Saturday') {
                if (numberOfPeople >= 100) {
                    totaPrice = numberOfPeople * 15.60;
                    discount = (numberOfPeople * 15.60) / 10;
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 15.60;
                }
            } else if (dayOfWeek === 'Sunday') {
                if (numberOfPeople >= 100) {
                    totaPrice = numberOfPeople * 16;
                    discount = (numberOfPeople * 16) / 10;
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 16;
                }
            }
            break;
        case 'Regular': 
            if (dayOfWeek === 'Friday') {
                if (numberOfPeople >= 10 && numberOfPeople <= 20) {
                    totaPrice = numberOfPeople * 15;
                    discount = totaPrice * (5 / 100);
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 15;
                }
            } else if (dayOfWeek === 'Saturday') {
                if (numberOfPeople >= 10 && numberOfPeople <= 20) {
                    totaPrice = numberOfPeople * 20;
                    discount = totaPrice * (5 / 100);
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 20;
                }
            } else if (dayOfWeek === 'Sunday') {
                if (numberOfPeople >= 10 && numberOfPeople <= 20) {
                    totaPrice = numberOfPeople * 22.50;
                    discount = totaPrice * (5 / 100);
                    price = totaPrice - discount;
                } else {
                    price = numberOfPeople * 22.50;
                }
            }   
            break;   
    }

    console.log(`Total price: ${price.toFixed(2)}`)
}

solve(30,
    "Students",
    "Sunday")

solve(40,
    "Regular",
    "Saturday")

File: Programming Fundamentals with Java/BASIC SYNTAX, CONDITIONAL STATEMENTS AND LOOPS/Vacation.java
import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        int group = Integer.parseInt(scanner.nextLine());
        String typeOfGroup = scanner.nextLine();
        String dayOfTheWeek = scanner.nextLine();

        double price = 0.0;
        double priceForTenPeople = 0.0;

        switch (dayOfTheWeek){
            case "Friday":
                if (typeOfGroup.equals("Students")){
                    price = group * 8.45;
                    if (group >= 30){
                        price -= price * (15.0f/100.0f);
                    }
                }else if (typeOfGroup.equals("Business")){
                    priceForTenPeople = 10.90 * 10;
                    price = group * 10.90;
                    if (group >= 100){
                        price -= priceForTenPeople;
                    }
                }else if (typeOfGroup.equals("Regular")){
                    price = group * 15;
                    if (group >= 10 && group <= 20){
                        price -= price * (5.0f/100.0f);
                    }
                }
                break;


            case "Saturday":
                if (typeOfGroup.equals("Students")){
                    price = group * 9.80;
                    if (group >= 30){
                        price -= price * (15.0f/100.0f);
                    }
                }else if (typeOfGroup.equals("Business")){
                    priceForTenPeople = 10 * 15.60;
                    price = group * 15.60;
                    if (group >= 100){
                        price -= priceForTenPeople;
                    }
                }else if (typeOfGroup.equals("Regular")){
                    price = group * 20;
                    if (group >= 10 && group <= 20){
                        price -= price * (5.0f/100.0f);
                    }
                }
                break;


            case "Sunday":
                if (typeOfGroup.equals("Students")){
                    price = group * 10.46;
                    if (group >= 30){
                        price -= price * (15.0f/100.0f);
                    }
                }else if (typeOfGroup.equals("Business")){
                    priceForTenPeople = 10 * 16;
                    price = group * 16;
                    if (group >= 100){
                        price -= priceForTenPeople;
                    }
                }else if (typeOfGroup.equals("Regular")){
                    price = group * 22.50;
                    if (group >= 10 && group <= 20){
                        price -= price * (5.0f/100.0f);
                    }
                }
                break;
        }
        System.out.printf("Total price: %.2f", price);
    }
}
